src/io/contsetch.c

```diff
@@ -21,3 +21,35 @@ s32 osContSetCh(u8 ch) {
     __osSiRelAccess();
     return ret;
 }
+
+/*
+ * Use a mask to detect controller channels instead, allowing the PIF to skip channels instead of
+ * spending time reading data.
+ * This does not work on iQue Player.
+ */
+s32 osContSetMask(u8 ch) {
```

Comment:
I think `osContSetCh` should probably be updated to call this function internally, enabling up to `ch` controllers. Something like
```c
s32 osContSetCh(u8 ch) {
    return osContSetMask((1 << ch) - 1);
}
```